crypto/key.go

```diff
@@ -66,6 +67,12 @@ func (k Key) isValidAtTime(t time.Time) bool {
 	return true
 }
 
+// MarshalJSON implements the json.Marshaler interface
+func (k Key) MarshalJSON() ([]byte, error) {
```

Comment:
Do we want to rename it to `MarshalKey`, if it is a specific thing OR if it is a common utility kind of thing, then probably move it to a common utils kinda file?

Reply:
It has to be called MarshalJSON: https://pkg.go.dev/encoding/json#Marshaler

essentially what this method does is overriding what json.Marshal does. Without it, the result is a dictionary with the key “Key” and the JWK inside that. That’s incorrect. This overrides the behavior so when the object is marshaled as JSON, it only marshals the value of the Key property.